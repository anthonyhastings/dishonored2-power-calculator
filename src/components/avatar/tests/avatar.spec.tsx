import { render, screen } from '@testing-library/react';
import Avatar, { type AvatarProps } from '../avatar';

describe('Avatar component', () => {
  let testContext: { defaultProps: AvatarProps };

  const renderComponent = (props = {}) =>
    render(<Avatar {...testContext.defaultProps} {...props} />);

  beforeEach(() => {
    testContext = {
      defaultProps: {
        name: 'Fake Name',
        slug: 'corvo',
      },
    };
  });

  describe('when name/slug is corvo', () => {
    beforeEach(() => {
      renderComponent({ name: 'Corvo', slug: 'corvo' });
    });

    it('renders an image specific to Corvo', () => {
      expect(screen.getByAltText('Portrait of Corvo')).toBeVisible();
    });
  });

  describe('when name/slug is emily', () => {
    beforeEach(() => {
      renderComponent({ name: 'Emily', slug: 'emily' });
    });

    it('renders an image specific to Emily', () => {
      expect(screen.getByAltText('Portrait of Emily')).toBeVisible();
    });
  });

  describe('when className is provided', () => {
    it('renders with extra className', () => {
      const { container } = renderComponent({ className: 'mock-class' });
      expect(container.querySelectorAll('.mock-class').length).toEqual(1);
    });
  });
});
