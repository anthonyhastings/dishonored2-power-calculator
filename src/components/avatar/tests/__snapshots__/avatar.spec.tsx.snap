// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component when className is provided renders with extra className 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ -1,8 +1,8 @@
  <img
    alt="Portrait of Fake Name"
-   className="avatar"
+   className="avatar mock-class"
    draggable="false"
    height="901"
    loading="lazy"
    src="corvo.jpg"
    width="771"
`;

exports[`Avatar component when slug is corvo renders an image specific to Corvo 1`] = `
<img
  alt="Portrait of Fake Name"
  className="avatar"
  draggable="false"
  height="901"
  loading="lazy"
  src="corvo.jpg"
  width="771"
/>
`;

exports[`Avatar component when slug is emily renders an image specific to Emily 1`] = `
<img
  alt="Portrait of Fake Name"
  className="avatar"
  draggable="false"
  height="901"
  loading="lazy"
  src="emily.jpg"
  width="771"
/>
`;
